plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal/Completion/CabalFields.hs

```diff
@@ -46,6 +50,71 @@ findFieldSection cursor (x:y:ys)
   where
     cursorLine = Syntax.positionRow cursor
 
+-- | Determine the field line the cursor is currently a part of.
+--
+-- The result is said field line and its starting position
+-- or Nothing if the passed list of fields is empty.
+
+-- This function assumes that elements in a field's @FieldLine@ list
+-- do not share the same row.
+findFieldLine :: Syntax.Position -> [Syntax.Field Syntax.Position] -> Maybe (Syntax.FieldLine Syntax.Position)
+findFieldLine _cursor [] = Nothing
+findFieldLine cursor fields =
+  case findFieldSection cursor fields of
+    Nothing                          -> Nothing
+    Just (Syntax.Field _ fieldLines) -> find filterLineFields fieldLines
+    Just (Syntax.Section _ _ fields) -> findFieldLine cursor fields
+  where
+    cursorLine = Syntax.positionRow cursor
+    -- In contrast to `Field` or `Section`, `FieldLine` must have the exact
+    -- same line position as the cursor.
+    filterLineFields (Syntax.FieldLine pos _) = Syntax.positionRow pos == cursorLine
+
+-- | Determine the exact word at the current cursor position.
+--
+-- The result is said word or Nothing if the passed list is empty
+-- or the cursor position is not next to, or on a word.
+-- For this function, a word is a sequence of consecutive characters
+-- that are not a space or column.
+
+-- This function currently only considers words inside of a @FieldLine@.
```

Comment:
```suggestion
-- that are not a space or column.
--
-- This function currently only considers words inside of a @FieldLine@.
```

Otherwise, the haddock comment will not be attached to the function :)